### DISCUSSION TITLE
esp32-2432S022C/N now also working!

### DISCUSSION THREAD

**Discussion by rzeldent** [rzeldent/esp32-smartdisplay#128] [Announcements]
The esp32-2432S022C/N has a different interface than all the other boards: a 8 bit 8080 interface.
This was there for a long time but not working; today found the issue and is working!

![PXL_20240210_130005037](https://github.com/rzeldent/esp32-smartdisplay/assets/7620231/92c6e62f-dc23-4769-945b-a5e26edee67f)

Enjoy!

**Comment by DAD451** [comment]
Hello, not work((

![WhatsApp Image 2024-11-06 at 15 21 22](https://github.com/user-attachments/assets/c99df6bb-0e13-4d7f-bdd0-6373ca7fcd92)
ELF file SHA256: 01ab03a94a319477

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4
[  5038][I][main.cpp:27] setup(): Board: esp32-2432S022C
[  5039][I][main.cpp:28] setup(): CPU: ESP32-D0WDQ5 rev3, CPU Freq: 240 Mhz, 2 core(s)
[  5040][I][main.cpp:29] setup(): Free heap: 267528 bytes
[  5045][I][main.cpp:30] setup(): Free PSRAM: 0 bytes
[  5050][I][main.cpp:31] setup(): SDK version: v4.4.3
E (5168) gpio: gpio_set_level(226): GPIO output gpio_num error
[  5184][E][esp_touch_cst816s.c:93] cst816s_reset(): Setting RST failed
[  5184][E][esp_touch_cst816s.c:265] esp_lcd_touch_new_i2c_cst816s(): GT911 reset failed
ESP_ERROR_CHECK failed: esp_err_t 0x102 (ESP_ERR_INVALID_ARG) at 0x400887e0
file: ".pio/libdeps/esp32-2432S022C/esp32_smartdisplay/src/lvgl_touch_cst816s_i2c.c" line 77
func: lvgl_touch_init
expression: esp_lcd_touch_new_i2c_cst816s(io_handle, &touch_config, &touch_handle)

abort() was called at PC 0x400887e3 on core 1


Backtrace: 0x40083b7d:0x3ffb2640 0x400887ed:0x3ffb2660 0x4008da75:0x3ffb2680 0x400887e3:0x3ffb2700 0x400feb7f:0x3ffb2720 0x400fe8e3:0x3ffb27b0 0x400d18ae:0x3ffb27d0 0x40100f92:0x3ffb2820

  #0  0x40083b7d:0x3ffb2640 in panic_abort at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_system/panic.c:402
  #1  0x400887ed:0x3ffb2660 in esp_system_abort at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_system/esp_system.c:128
  #2  0x4008da75:0x3ffb2680 in abort at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/newlib/abort.c:46
  #3  0x400887e3:0x3ffb2700 in _esp_error_check_failed at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_system/esp_err.c:43
  #4  0x400feb7f:0x3ffb2720 in lvgl_touch_init at .pio/libdeps/esp32-2432S022C/esp32_smartdisplay/src/lvgl_touch_cst816s_i2c.c:47 (discriminator 1)
  #5  0x400fe8e3:0x3ffb27b0 in smartdisplay_init at .pio/libdeps/esp32-2432S022C/esp32_smartdisplay/src/esp32_smartdisplay.c:210
  #6  0x400d18ae:0x3ffb27d0 in setup() at src/main.cpp:33
  #7  0x40100f92:0x3ffb2820 in loopTask(void*) at C:/Users/DAD/.platformio/packages/framework-arduinoespressif32/cores/esp32/main.cpp:42

**Reply by DAD451** [reply]
```
 // Create touch configuration
    const esp_lcd_touch_config_t touch_config = {
        .x_max = CST816S_TOUCH_CONFIG_X_MAX,
        .y_max = CST816S_TOUCH_CONFIG_Y_MAX,
        //.rst_gpio_num = CST816S_TOUCH_CONFIG_RST_GPIO_NUM,
        //.int_gpio_num = CST816S_TOUCH_CONFIG_INT_GPIO_NUM,
        .levels = {
            .reset = CST816S_TOUCH_CONFIG_LEVELS_RESET,
            .interrupt = CST816S_TOUCH_CONFIG_LEVELS_INTERRUPT},
        .flags = {.swap_xy = TOUCH_SWAP_XY, .mirror_x = TOUCH_MIRROR_X, .mirror_y = TOUCH_MIRROR_Y},
        .user_data = io_handle};
    log_d("touch_config: x_max:%d, y_max:%d, rst_gpio_num:%d, int_gpio_num:%d, levels:{reset:%d, interrupt:%d}, flags:{swap_xy:%d, mirror_x:%d, mirror_y:%d}, user_data:0x%08x", touch_config.x_max, touch_config.y_max, touch_config.rst_gpio_num, touch_config.int_gpio_num, touch_config.levels.reset, touch_config.levels.interrupt, touch_config.flags.swap_xy, touch_config.flags.mirror_x, touch_config.flags.mirror_y, touch_config.user_data);
    esp_lcd_touch_handle_t touch_handle;
    ESP_ERROR_CHECK(esp_lcd_touch_new_i2c_cst816s(io_handle, &touch_config, &touch_handle));
```
I commented rst_gpio_num and int_gpio_num, now all work. 
![e7279a31-7311-4527-a74b-a980a94dedff](https://github.com/user-attachments/assets/1f543fa9-9adb-4b51-8d72-2ef021e03b74)

**Comment by rzeldent** [comment]
Yes, you're right. In the driver there is no check to see if the RST is NC

**Reply by DAD451** [reply]
If you could get the Openhasp project up and running with its own driver, that would be awesome!
https://github.com/HASwitchPlate/openHASP/issues/606
